### DISCUSSION TITLE
线上业务，客户端和服务端如何快速升级官方新版本？升级方案是什么样的

### DISCUSSION THREAD

**Discussion by painAdmin** [apolloconfig/apollo#5250] [Q&A]
背景：
当前配置中心，有大量的服务在使用，目前有3个环境：测试，灰度，线上，如果快速的升级官方新版本，使用新的功能？如何减少对使用方的影响，或减少使用方的改动成本？

问题：
1.看了其他评论，建议是客户端和服务端版本一致，那如果服务端升级，是否对客户端有影响？如果没响应，那么没问题，有影响，是不是代表，后续新的版本都无法使用了

2. 升级服务端对旧的客户端兼容，这个该怎么保证，只有release note和单侧，无法完全排除风险？

建议：
1.官方给出明确的服务端版本，兼容客户端版本的列表，让大家无忧升级
2.对于确实受影响的客户端版本，官方给出推荐的升级方案，方便大家操作

**Comment by nobodyiam** [comment]
1. 服务端版本大于等于客户端版本是没问题的。
2. 如果客户端版本大于服务端版本，那就要自己看下 release note 了。
